projects/batfish/src/main/java/org/batfish/representation/juniper/JuniperConfiguration.java

```diff
@@ -247,6 +257,9 @@ private BgpProcess createBgpProcess(RoutingInstance routingInstance) {
       IpBgpGroup ig = e.getValue();
       BgpNeighbor neighbor = new BgpNeighbor(ip, _c);
       neighbor.setVrf(vrfName);
+      neighbor.setAuthAlgorithm(ig.getAuthAlgorithm());
+      neighbor.setAuthKey(ig.getAuthKey());
+      neighbor.setAuthKeyChainName(ig.getAuthKeyChainName());
```

Comment:
Verify the `keyChainName` actually exists in `_authenticationKeyChains` here? What to do if there is no such `keyChainName` defined?